From 687aed860919ed3077b60e7d9e9decf50c5c8d6c Mon Sep 17 00:00:00 2001
From: Wilson Ding <dingwei@marvell.com>
Date: Wed, 30 Mar 2016 13:13:58 +0800
Subject: [PATCH 0998/1240] fix: env: update the console name according to
 selected uart driver

- ARAMADA3700 integrates a new UART IP. Its Linux driver uses 'ttyMV0'
  as console name in Kernel 4.4.1. However, the current u-Boot assumes
  all MVEBU SoCs uses NS16550 UART driver and choose 'ttyS0' as the
  default console name.
- To solve the conflicts between the platforms, CONFIG_DEFAULT_CONSOLE
  is introduced to the config file. It will choose the console name
  according to the UART driver selected.

Change-Id: I8ecc1419b044c293a3a71baaff5ed32352a2aac2
Signed-off-by: Wilson Ding <dingwei@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/28653
Reviewed-by: Hua Jing <jinghua@marvell.com>
Tested-by: Star_Automation <star@marvell.com>
Reviewed-by: Haim Boot <hayim@marvell.com>
---
 include/configs/mvebu-common.h | 7 ++++++-
 1 file changed, 6 insertions(+), 1 deletion(-)

diff --git a/include/configs/mvebu-common.h b/include/configs/mvebu-common.h
index f5e25c1..e7b5fad 100644
--- a/include/configs/mvebu-common.h
+++ b/include/configs/mvebu-common.h
@@ -162,7 +162,7 @@
 					"$ramfs_name; else setenv ramfs_addr -;fi\0"				\
 					"get_images=tftp $kernel_addr $image_name; tftp $fdt_addr $fdt_name; "	\
 						"run get_ramfs\0"						\
-					"console=console=ttyS0,115200\0"					\
+					"console=" CONFIG_DEFAULT_CONSOLE "\0"					\
 					"root=root=/dev/nfs rw\0"						\
 					"set_bootargs=setenv bootargs $console $root "				\
 						"ip=$ipaddr:$serverip:$gatewayip:$netmask:$hostname:$netdev:none "\
@@ -272,8 +272,13 @@
 	#define CONFIG_CONS_INDEX		1
 	#define CONFIG_SYS_LOADS_BAUD_CHANGE	/* allow baudrate changes */
 	/*#define CONFIG_SYS_DUART_CHAN		0*/
+	#define CONFIG_DEFAULT_CONSOLE		"console=ttyS0,115200"
 #endif /* CONFIG_SYS_NS16550 */
 
+#ifdef CONFIG_MVEBU_UART_ARLP
+	#define CONFIG_DEFAULT_CONSOLE		"console=ttyMV0,115200 earlycon=ar3700_uart,0xd0012000"
+#endif
+
 #if defined(CONFIG_SYS_NS16550) || defined(CONFIG_MVEBU_UART_ARLP)
 #define CONFIG_BAUDRATE		115200
 #endif
-- 
1.9.1

